src/util/color/rgbcolor.h

```diff
@@ -64,22 +64,26 @@ class RgbColor {
     static constexpr optional_t optional(RgbColor color) {
         return std::make_optional(color);
     }
-    static optional_t optional(QColor color) {
+    static optional_t optional(
+            QColor color,
+            optional_t defaultColor = nullopt()) {
         if (color.isValid()) {
             return optional(validateCode(color.rgb()));
         } else {
-            return nullopt();
+            return defaultColor;
         }
     }
-    static optional_t optional(const QVariant& varCode) {
+    static optional_t optional(
+            const QVariant& varCode,
```

Comment:
Since we do have a `toQString()` function it would make sense to add something like this, too:

```c++
static optional_t optional(const QString& strCode, optional_t defaultColor = nullopt());
```

Reply:
We need to decide where to draw the line. If we parse QString then what about if a QVariant contains a string instead of an uint? Should we silently try to parse it?? I don't think that this is a good idea.

This seems to get out of control. Any ideas where to locate all the desired functionality and prevent this innocent little class from becoming a behemoth?

Reply:
Parsing a string and an integer are fundamentally different. I need to think about the requirements to come up with a proposal. The function names should reveal what is intended. And we should extract all extended conversions into a separate module, consistently as free utility functions.

Reply:
Agreed.

Reply:
They will become `static` functions of `RgbColor`. Most convenient to use/read and still somehow separate from the actual class.

Reply:
Otherwise we would need to place everything in a separate namespace to allow short functions names without risking name clashes.

Reply:
Done. Includes now hopefully all conversions between RgbColor and Qt types that we'll (ever) need.